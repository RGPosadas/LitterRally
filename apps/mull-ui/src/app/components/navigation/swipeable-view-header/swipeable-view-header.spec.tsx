import { render } from '@testing-library/react';
import { createMemoryHistory } from 'history';
import React from 'react';
import renderer from 'react-test-renderer';
import SwipeableViewHeader from './swipeable-view-header';

describe('SwipeableViewHeader', () => {
  it('should render successfully', () => {
    const { baseElement } = render(
      <SwipeableViewHeader index={0} setIndex={jest.fn()} viewTitles={['Slide1', 'Slide2']} />
    );

    expect(baseElement).toBeTruthy();
  });

  it('should have the correct button active based the view', () => {
    const utils = render(
      <SwipeableViewHeader index={0} setIndex={jest.fn()} viewTitles={['Slide1', 'Slide2']} />
    );
    utils.getByTestId('subnavigation-slide1-button').click();
    utils.rerender(
      <SwipeableViewHeader index={1} setIndex={jest.fn()} viewTitles={['Slide1', 'Slide2']} />
    );
    const subnavHeader = utils.getByTestId('subnavigation-slide2-button');
    expect(subnavHeader.classList.contains('active'));
  });

  it('should match snapshot', () => {
    const history = createMemoryHistory();
    const tree = renderer
      .create(
        <SwipeableViewHeader index={0} setIndex={jest.fn()} viewTitles={['Slide1', 'Slide2']} />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
