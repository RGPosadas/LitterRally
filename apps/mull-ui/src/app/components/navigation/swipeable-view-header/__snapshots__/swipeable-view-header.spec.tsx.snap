// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwipeableViewHeader should match snapshot 1`] = `
<div
  className="sub-nav-bar-container undefined"
>
  <div
    className="inner-cont"
  >
    <p
      className="subnavigation-link active"
      data-testid="subnavigation-slide1-button"
      onClick={[Function]}
    >
      Slide1
    </p>
    <p
      className="subnavigation-link "
      data-testid="subnavigation-slide2-button"
      onClick={[Function]}
    >
      Slide2
    </p>
  </div>
</div>
`;
